mlrun/db/httpdb.py

```diff
@@ -330,38 +347,50 @@ def _api_call(_params):
                 version=version,
             )
 
-        first_page_params = deepcopy(params) or {}
-        first_page_params["page"] = 1
-        first_page_params["page-size"] = config.httpdb.pagination.default_page_size
-        response = _api_call(first_page_params)
+        page_params = deepcopy(params) or {}
+
+        if page_params.get("page-token") is None and page_params.get("page") is None:
+            page_params["page"] = 1
+
+        if page_params.get("page-size") is None:
+            page_params["page-size"] = config.httpdb.pagination.default_page_size
+
+        response = _api_call(page_params)
 
+        # Yield only a single page of results
         yield response
-        page_token = response.json().get("pagination", {}).get("page-token", None)
-
-        while page_token:
-            try:
-                # Use the page token to get the next page.
-                # No need to supply any other parameters as the token informs the pagination cache
-                # which parameters to use.
-                response = _api_call({"page-token": page_token})
-            except mlrun.errors.MLRunNotFoundError:
-                # pagination token expired
-                break
-
-            yield response
+
+        if return_all:
             page_token = response.json().get("pagination", {}).get("page-token", None)
 
+            while page_token:
+                try:
+                    # Use the page token to get the next page.
+                    # No need to supply any other parameters as the token informs the pagination cache
+                    # which parameters to use.
+                    response = _api_call({"page-token": page_token})
+                except mlrun.errors.MLRunNotFoundError:
+                    # pagination token expired
+                    break
```

Comment:
Do we want to inform a user in this case?

Reply:
@rokatyy No, it simply means that we've reached the last page (I updated the comment).